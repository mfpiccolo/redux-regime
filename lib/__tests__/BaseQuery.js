"use strict";

var _BaseModel2 = require("../BaseModel");

var _BaseModel3 = _interopRequireDefault(_BaseModel2);

var _jsonApiNormalizer = require("json-api-normalizer");

var _jsonApiNormalizer2 = _interopRequireDefault(_jsonApiNormalizer);

var _graphqlNormalizr = require("graphql-normalizr");

var _checklistsAndTasksNormalized = require("../fixtrues/checklistsAndTasksNormalized");

var _checklistsAndTasksNormalized2 = _interopRequireDefault(
  _checklistsAndTasksNormalized
);

function _interopRequireDefault(obj) {
  return obj && obj.__esModule ? obj : { default: obj };
}

function _classCallCheck(instance, Constructor) {
  if (!(instance instanceof Constructor)) {
    throw new TypeError("Cannot call a class as a function");
  }
}

function _possibleConstructorReturn(self, call) {
  if (!self) {
    throw new ReferenceError(
      "this hasn't been initialised - super() hasn't been called"
    );
  }
  return call && (typeof call === "object" || typeof call === "function")
    ? call
    : self;
}

function _inherits(subClass, superClass) {
  if (typeof superClass !== "function" && superClass !== null) {
    throw new TypeError(
      "Super expression must either be null or a function, not " +
        typeof superClass
    );
  }
  subClass.prototype = Object.create(superClass && superClass.prototype, {
    constructor: {
      value: subClass,
      enumerable: false,
      writable: true,
      configurable: true
    }
  });
  if (superClass)
    Object.setPrototypeOf
      ? Object.setPrototypeOf(subClass, superClass)
      : (subClass.__proto__ = superClass);
}

var Checklist = (function(_BaseModel) {
  _inherits(Checklist, _BaseModel);

  function Checklist() {
    _classCallCheck(this, Checklist);

    return _possibleConstructorReturn(
      this,
      (Checklist.__proto__ || Object.getPrototypeOf(Checklist))
        .apply(this, arguments)
    );
  }

  return Checklist;
})(_BaseModel3.default);

Checklist.hasMany = ["tasks"];

describe("BaseModel", function() {
  test("query returns a QueryObject", function() {
    expect(_BaseModel3.default.query()).toBeInstanceOf(_BaseModel2.QueryObject);
  });

  describe("all()", function() {
    test("", function() {
      expect(
        Checklist.query(_checklistsAndTasksNormalized2.default).all().execute()
      ).toMatchSnapshot();
    });
  });

  describe("find()", function() {
    test("", function() {
      var checklist = Checklist.query(_checklistsAndTasksNormalized2.default)
        .find(1)
        .execute()[0];
      expect(checklist.id).toEqual(1);
      expect(checklist).toMatchSnapshot();
    });
  });

  describe("where()", function() {
    test("", function() {
      var checklist = Checklist.query(_checklistsAndTasksNormalized2.default)
        .where({ name: "Project Audit Rest" })
        .execute()[0];
      expect(checklist.id).toEqual(2);
      expect(checklist).toMatchSnapshot();
    });
  });

  describe("includes()", function() {
    test("", function() {
      var checklists = Checklist.query(_checklistsAndTasksNormalized2.default)
        .all()
        .includes(["tasks"])
        .execute();
      expect(checklists).toMatchSnapshot();
    });
  });

  describe("hasMany()", function() {
    test("", function() {
      var checklists = Checklist.query(_checklistsAndTasksNormalized2.default)
        .tasks()
        .all()
        .execute();
      expect(checklists).toMatchSnapshot();
    });
  });
});
