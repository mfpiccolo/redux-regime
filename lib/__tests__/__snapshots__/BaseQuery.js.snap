// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseModel all()  1`] = `
Array [
  Object {
    "id": 1,
    "name": "Onboarding Rest",
  },
  Object {
    "id": 2,
    "name": "Project Audit Rest",
  },
  Object {
    "id": 3,
    "name": "QA Checklist Rest",
  },
]
`;

exports[`BaseModel find()  1`] = `
Object {
  "id": 1,
  "name": "Onboarding Rest",
}
`;

exports[`BaseModel hasMany()  1`] = `
Array [
  Object {
    "description": "Onboarding REST 1",
    "id": 1,
  },
  Object {
    "description": "Onboarding REST 2",
    "id": 2,
  },
  Object {
    "description": "Project Audit Rest 3",
    "id": 3,
  },
  Object {
    "description": "Project Audit Rest 4",
    "id": 4,
  },
  Object {
    "description": "QA Checklist Rest 5",
    "id": 5,
  },
  Object {
    "description": "QA Checklist Rest 6",
    "id": 6,
  },
]
`;

exports[`BaseModel includes()  1`] = `
Array [
  Object {
    "id": 1,
    "name": "Onboarding Rest",
    "tasks": Array [
      Object {
        "description": "Onboarding REST 1",
        "id": 1,
        "type": "tasks",
      },
      Object {
        "description": "Onboarding REST 2",
        "id": 2,
        "type": "tasks",
      },
    ],
  },
  Object {
    "id": 2,
    "name": "Project Audit Rest",
    "tasks": Array [
      Object {
        "description": "Project Audit Rest 3",
        "id": 3,
        "type": "tasks",
      },
      Object {
        "description": "Project Audit Rest 4",
        "id": 4,
        "type": "tasks",
      },
    ],
  },
  Object {
    "id": 3,
    "name": "QA Checklist Rest",
    "tasks": Array [
      Object {
        "description": "QA Checklist Rest 5",
        "id": 5,
        "type": "tasks",
      },
      Object {
        "description": "QA Checklist Rest 6",
        "id": 6,
        "type": "tasks",
      },
    ],
  },
]
`;

exports[`BaseModel where()  1`] = `
Object {
  "id": 2,
  "name": "Project Audit Rest",
}
`;
